signingscript/src/signingscript/sign.py

```diff
@@ -1512,7 +1536,38 @@ async def rcodesign_notarize(paths, notarization_workdir, creds_path):
             creds_path,
             app_path,
         ]
-        await utils.execute_subprocess(command)
+
+        # Similar implementation to utils.execute_subprocess, but handling some errors:
+        message = 'Running "{}"'.format(" ".join(command))
+        log.info(message)
+        subprocess = await asyncio.create_subprocess_exec(
+            *command,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.STDOUT,
+        )
+        log.info("COMMAND OUTPUT: ")
+        output_lines = []
+        while True:
+            line = await subprocess.stdout.readline()
+            if not line:
+                break
+            line = line.decode("utf-8").rstrip()
+            log.info(line)
+            output_lines.append(line)
+        exitcode = await subprocess.wait()
+        log.info("exitcode {}".format(exitcode))
+        if exitcode > 0:
+            if not output_lines or "Connection reset by peer" not in output_lines[-1]:
+                raise SigningScriptError("Unknown error from notarization service.")
+            log.info("Connection reset by peer, attempting to recover")
+            for line in output_lines:
+                if "submission ID: " in line:
+                    submission_id = line.split(": ")[-1]
+                    break
+            else:
+                raise SigningScriptError("Unable to find notarization submission ID from logs.")
+            await rcodesign_notary_wait(submission_id, creds_path)
+            await rcodesign_staple(app_path)
```

Comment:
If I understand how all of the `rcodesign` commands work correctly, it looks like we may end up either stapling in the original `notary-submit` command, or down here. I'd strongly suggest that you change this, and make the original command only submit, so that we're always stapling in exactly the same way. This will both make it easier to understand the flow of things now, and also avoid future pitfalls (such as stapling potentially being done with different arguments if we ever make changes there). It may also make it easier to parallelize in the future, since submission will no longer be a long lived subprocess.